examples/vision/forwardforward.py

```diff
@@ -0,0 +1,352 @@
+training on each layer as per the number of epochs per layer.
+"""
+
+
+class FFNetwork(keras.Model):
+    def __init__(self, dims, layer_optimizer=keras.optimizers.Adam(learning_rate=0.03)):
+        super().__init__()
+        self.layer_optimizer = layer_optimizer
+        self.mean_loss = keras.metrics.Mean()
+        self.flatten_layer = keras.layers.Flatten()
+        self.layer_list = [keras.Input(shape=(dims[0],))]
+        for d in range(len(dims) - 1):
+            self.layer_list += [FFDense(dims[d + 1], optimizer=self.layer_optimizer)]
+
+    @tf.function()
+    def overlay_y_on_x(self, X, y):
+        x_res = X.numpy()
+        x_npy = X.numpy()
+        x_res[:, :10] *= 0.0
+        if not isinstance(y, int):
+            y_npy = y.numpy()
+            x_res[range(x_npy.shape[0]), y.numpy()] = x_npy.max()
+        else:
+            x_res[range(x_npy.shape[0]), y] = x_npy.max()
+        return tf.convert_to_tensor(x_res)
+
+    @tf.function()
+    def predict_one_sample(self, x):
+        goodness_per_label = []
+        x = tf.expand_dims(x, axis=0)
+        for label in range(10):
+            h = self.overlay_y_on_x(x, label)
+            h = self.flatten_layer(h)
+            goodness = []
+            for layer_idx in range(1, len(self.layer_list)):
+                layer = self.layer_list[layer_idx]
+                h = layer(h)
+                goodness += [tf.math.reduce_mean(tf.math.pow(h, 2), 1)]
+            goodness_per_label += [
+                tf.expand_dims(tf.reduce_sum(goodness, keepdims=True), 1)
+            ]
+        goodness_per_label = tf.concat(goodness_per_label, 1)
+        return tf.argmax(goodness_per_label, 1)
+
+    def predict(self, data):
+        x = data
+        preds = list()
+        for idx in trange(x.shape[0]):
+            sample = x[idx]
+            result = self.predict_one_sample(sample)
+            preds.append(result)
+        return np.asarray(preds, dtype=int)
+
+    def train_step(self, data):
+        x, y = data
+        x = self.flatten_layer(x)
+        perm_array = tf.range(start=0, limit=x.get_shape()[0], delta=1)
+        x_pos = self.overlay_y_on_x(x, y)
+        y_numpy = y.numpy()
+        random_y_tensor = y_numpy[tf.random.shuffle(perm_array)]
+        x_neg = self.overlay_y_on_x(x, tf.convert_to_tensor(random_y_tensor))
+        h_pos, h_neg = x_pos, x_neg
+        for idx, layer in enumerate(self.layers):
+            if idx == 0:
+                print("Input layer : No training")
+                continue
+            print(f"Training layer {idx+1} now : ")
+            if isinstance(layer, FFDense):
+                h_pos, h_neg, loss, loss_list = layer.forwardforward(h_pos, h_neg)
+                plt.plot(range(len(loss_list)), loss_list)
+                plt.title(f"Loss over training on layer {idx+1}")
+                plt.show()
+            else:
+                x = layer(x)
+        return {"FinalLoss": loss}
```

Comment:
`train_step` should return average values, e.g. the output of `loss_tracker.result()` (where `loss_tracker` is a `metrics.Mean` instance).

Also, use snake case

Reply:
Will make the change